lnwire/onion_error_test.go

```diff
@@ -189,8 +187,32 @@ func TestFailIncorrectDetailsOptionalAmount(t *testing.T) {
 		t.Fatalf("unable to decode error: %v", err)
 	}
 
-	if !reflect.DeepEqual(onionError, onionError2) {
-		t.Fatalf("expected %v, got %v", spew.Sdump(onionError),
-			spew.Sdump(onionError2))
+	invalidDetailsErr, ok := onionError2.(*FailIncorrectDetails)
+	if !ok {
+		t.Fatalf("expected FailIncorrectDetails, but got %T",
+			onionError2)
 	}
+
+	if invalidDetailsErr.amount != 0 {
+		t.Fatalf("expected amount to be zero")
+	}
+}
+
+type legacyFailIncorrectDetailsNoAmt struct {
```

Comment:
`mockFailIncorrectDetailsNoAmte`?

Reply:
It really is the legacy failure that no longer exists, but we can still receive it.

Reply:
Since methods are empty, and this is only in test code, looks more like a mock to me than an actual legacy thing :P

Reply:
Renamed